### DISCUSSION TITLE
Agent/Process Frameworks for Complex Use Cases

### DISCUSSION THREAD

**Discussion by neuralcoderj** [microsoft/semantic-kernel#9780] [Q&A] [answered]
Hello,

I’d like to clarify about agents and the process framework in Semantic Kernel. Is the reason behind the emergence of these frameworks due to the difficulty of orchestrating plugins by the LLM and its function calling capability as the number of plugins increases? Up until now, I have experienced Semantic Kernel without the agent framework, meaning with a single kernel and a maximum of 3-4 plugins, with a total of 15-16 kernel functions across them, and I found its performance sufficient. However, let's say that our use case becomes more complex, and we need to increase the number of plugins to 15-20, and the number of functions to 40-50. In this case, I’m curious about the orchestration performance of the LLM. In such a scenario, can the function calling capability still come up with the right plan and take the correct actions? Or would you recommend integrating the agent framework with the process framework to build a more powerful and resilient structure?

**Comment by sophialagerkranspandey** [comment, marked as answer]
Absolutely, your thinking is spot on. We would recommend integrating Agent Framework with the Process Framework to build a resilient and powerful structure. Adding @alliscode and @sphenry for any further elaboration. Thanks!

**Comment by sharma18yash** [comment]
Do we have any example for integrating agent framework with process framework?

**Reply by moonbox3** [reply]
Are you using SK .NET or Python, @sharma18yash?

**Reply by sharma18yash** [reply]
I am using .NET SK

**Reply by diegomodolo** [reply]
I would like that example too!
